### ISSUE TITLE
Support for compressed secrets

### ISSUE THREAD

**Issue by s-fleck** [gaborcsardi/secret#36]
Would it be possible to support compression for secrets (similar to the different compression algorithms supported by saveRDS?). I sometimes have to store sensitive datasets inside packages in a setting where file size matters (a bit).

**Comment by gaborcsardi** [comment]
Can you compress the file first, and then add it to the vault?

**Comment by s-fleck** [comment]
in my case it's a data.frame with confidentak

**Comment by gaborcsardi** [comment]
Yeah, it would make sense to compress it I guess. We could `saveRDS()` it into a connection in memory.
